src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -39,84 +39,73 @@
 using namespace chip;
 using namespace chip::app::Clusters::WindowCovering;
 
-#define WC_PERCENT100THS_MIN 0
-#define WC_PERCENT100THS_MAX 10000
-
-static bool HasFeature(chip::EndpointId endpoint, WcFeature feature)
-{
-    uint32_t FeatureMap = 0;
-    if (EMBER_ZCL_STATUS_SUCCESS ==
-        emberAfReadServerAttribute(endpoint, chip::app::Clusters::WindowCovering::Id,
-                                   chip::app::Clusters::WindowCovering::Attributes::FeatureMap::Id,
-                                   reinterpret_cast<uint8_t *>(&FeatureMap), sizeof(FeatureMap)))
-    {
-        return (FeatureMap & chip::to_underlying(feature)) != 0;
-    }
-
-    return false;
-}
-
-static bool HasFeaturePaLift(chip::EndpointId endpoint)
-{
-    return (HasFeature(endpoint, WcFeature::kLift) && HasFeature(endpoint, WcFeature::kPositionAwareLift));
-}
-
-static bool HasFeaturePaTilt(chip::EndpointId endpoint)
-{
-    return (HasFeature(endpoint, WcFeature::kTilt) && HasFeature(endpoint, WcFeature::kPositionAwareTilt));
-}
-
-static uint16_t ValueToPercent100ths(uint16_t openLimit, uint16_t closedLimit, uint16_t value)
+#define CHECK_BOUNDS_INVALID(MIN, VAL, MAX) ((VAL < MIN) || (VAL > MAX))
+#define CHECK_BOUNDS_VALID(MIN, VAL, MAX) (!CHECK_BOUNDS_INVALID(MIN, VAL, MAX))
+
+/*
+ * ConvertValue: Permits to convert values from one range to another
+ * Range In  -> from  inputLowValue to   inputHighValue
+ * Range Out -> from outputLowValue to outputtHighValue
+ * offset true  -> allows to take into account the minimum of each range in the conversion
+ * offset false -> applies only the basic "rule of three" for the conversion
```

Comment:
I'm not sure what the "rule of three" is... But also, do we have any callers with offset == false?  Do we plan to have any?

Reply:
I just wanted to allows the usage of this conversion in the function "in case" but so far i don't use it

Reply:
I feel we should keep it in case otherwise someone would like to use it

And also I think this kind of helper function shouldn't be limited to window covering but available for the whole Matter codebase. Many clusters  certainly re-implement that over and over again IMO

Reply:
rule of three : is the basic computation you do for a percentage for example
not sure how it is called in English ..